### DISCUSSION TITLE
Can CoPilot use information from old commits from my repository?

### DISCUSSION THREAD

**Discussion by FireVolkhov** [community/community#60050] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Can CoPilot use information from old commits from my repository? Code that has been overwritten.

**Comment by Akash1134** [comment, marked as answer]
Hi @FireVolkhov ,

We announced a feature "[Referencing public code](https://github.blog/2022-11-01-preview-referencing-public-code-in-github-copilot/)" back in November as a preview that we're still working on. With that in place, within this context, we will begin to introduce several new capabilities to Copilot. For any code fragment suggested by Copilot, developers should have:
- An inventory of similar code found in GitHub public repositories
- The ability to sort that inventory by repository license, commit date, etc.

There is another exciting work in progress that'll let GitHub Copilot see your entire repo when it comes up with its suggestions. Specifically, showing Copilot snippets from my private codebase that are relevant to the task you're trying to accomplish. For details, see [GitHub Copilot for Your Codebase](https://githubnext.com/projects/copilot-view/)
